### PR TITLE
Add check for out of date build compared to source files. Offer rebui…

### PR DESCRIPTION
**Pull request by kevglass**
Adds a check to see if the source files are newer than the distribution files. If they are we offer to run a npm run build before doing the upload. 

There are some assumption about how the game directories are organised based on the `rune create` command. This need to be reviewed.

### CODE REVIEW

**Comment by kevglass** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by bfelbo** [comment]
Thanks! I'm sure this'll help prevent others from releasing game versions with bugs 💯

>  If they are we offer to run a npm run build before doing the upload.

Perhaps it'd be safer to prompt `⚠️ The files you're about to upload are older than your source files. You might need to build/compile your game before uploading. Are you sure you want to continue?`

I'm not sure we can make assumptions about what build system or commands people use to make their game.

**Comment by kevglass** [comment]
> Thanks! I'm sure this'll help prevent others from releasing game versions with bugs 💯
> 
> > If they are we offer to run a npm run build before doing the upload.
> 
> Perhaps it'd be safer to prompt `⚠️ The files you're about to upload are older than your source files. You might need to build/compile your game before uploading. Are you sure you want to continue?`
> 
> I'm not sure we can make assumptions about what build system or commands people use to make their game.

Done and done.

File: packages/rune-games-cli/src/flows/Upload/Upload.tsx [hunk 1]
```diff
@@ -96,9 +113,21 @@ export function Upload({ flags }: { flags: CliFlags }) {
                 />
               )}
 
+              {!!gameId && !buildCheckComplete && buildRequired(gameDir) && (
```
Comment (Gongreg) [inline packages/rune-games-cli/src/flows/Upload/Upload.tsx]: Lets move this before selecting the game, that way user will find out sooner that they need to run the build.

Comment (kevglass) [reply packages/rune-games-cli/src/flows/Upload/Upload.tsx]: Sorry for the delay, thats in now.

**Review by Gongreg — APPROVED** [review]

**Pull request merged**